### PR TITLE
Unify criteria management

### PR DESCRIPTION
**Pull request by GitGeminiano**
Hello everyone,

I merged the help panes in the page that includes the list of criteria. I also fixed some console errors and the update functionality when criteria are edited right after they are created. Finally, I also migrated some files from the flexible_criteria and rubric_criteria folders into a common criteria folder, which will facilitate the unification of both flexible and rubric criteria.

Regards,
Christian

### CODE REVIEW

File: app/views/criteria/_criteria_pane.html.erb [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
     </span>
     <span class='right'><%= t('visibility') %></span>
   </header>
 
-  <ul id='flexible_criteria_pane_list'>
+  <ul id='criteria_pane_list'>
     <% @criteria.each do |criterion| %>
-      <%= render partial: 'flexible_criteria/flexible_criterion',
-                 locals: { criterion: criterion } %>
+        <%= render partial: 'criteria/criterion',
```
Comment (david-yz-liu) [inline app/views/criteria/_criteria_pane.html.erb]: Undo the indentation change

File: app/views/criteria/_criterion.html.erb [hunk 2]
```diff
@@ -1,11 +1,14 @@
 <li id='criterion_<%= criterion.id %>'>
-  <span class='left'>
+  <span id class='left'>
```
Comment (david-yz-liu) [inline app/views/criteria/_criterion.html.erb]: There's no value for `id` here

File: app/views/criteria/_new_criterion.html.erb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<li id="new_criterion">
+	<%= form_for @criterion,
```
Comment (david-yz-liu) [inline app/views/criteria/_new_criterion.html.erb]: This should be indented only 2 spaces

File: app/views/criteria/_new_criterion.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<li id="new_criterion">
```
Comment (david-yz-liu) [inline app/views/criteria/_new_criterion.html.erb]: generally use single-quotes

Comment (GitGeminiano) [inline app/views/criteria/_new_criterion.html.erb]: Thanks. I will make the necessary changes.

File: app/views/criteria/_new_criterion.html.erb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+           action: 'index',
+           assignment_id: assignment.id },
+    remote: true do |f| %>
+    <div id="new_criterion_errors" style="display:none;" class="error"></div>
+    <%= label_tag 'new_criterion_prompt',
+		  t("criteria.name") %><br>
+    <%= f.text_field :name,
+		     size: 15,
+		     id: 'new_criterion_prompt' %><br>
+    <%= f.submit t("criteria.save"), :'data-disable-with' => I18n.t('working')%>
```
Comment (GitGeminiano) [inline app/views/criteria/_new_criterion.html.erb]: I will change this, too.

**Comment by david-yz-liu** [comment]
@GitGeminiano this is good, but I noticed that the criteria list has lost the ability to reorder criteria through click-and-drag, at least on Chrome. Please look into this issue.

**Comment by GitGeminiano** [comment]
I will look into it. Thanks.

**Pull request merged**